### PR TITLE
[Enhancement] Feature quick preview file

### PR DESCRIPTION
**Pull request by andrk0p**
### Description of Change ###

Added the ability to quickly view the file inside the iOS app.

### API Changes ###

Added: 
`public bool QuickLook { get; set; } = false;`
To
`OpenFileRequest`

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of `main` at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

File: Xamarin.Essentials/Launcher/Launcher.netstandard.watchos.cs [hunk 1]
```diff
@@ -14,6 +14,9 @@ public static partial class Launcher
         static Task PlatformOpenAsync(OpenFileRequest request) =>
             throw ExceptionUtils.NotSupportedOrImplementedException;
 
+        static Task PlatformOpenAsync(OpenFileRequest request, bool openInApp) =>
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/Launcher/Launcher.netstandard.watchos.cs:17]: No need for another parameter, it should just be in the OpenFileRequest

**Review by jamesmontemagno — CHANGES_REQUESTED** [review]
I would only add it to the OpenFileRequest. This simplifies the API  so we don't need a new one and can document that it is only an iOS feature.

File: Xamarin.Essentials/Launcher/Launcher.android.cs [hunk 2]
```diff
@@ -57,6 +57,9 @@ static Task PlatformOpenAsync(OpenFileRequest request)
             return Task.CompletedTask;
         }
 
+        static Task PlatformOpenAsync(OpenFileRequest request, bool openInApp) =>
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/Launcher/Launcher.android.cs:60]: I think we would just add this to "OpenFileReqauest", no need to chagne teh singautre at all

File: Xamarin.Essentials/Launcher/Launcher.ios.tvos.cs [hunk 3]
```diff
@@ -62,9 +65,60 @@ static Task PlatformOpenAsync(OpenFileRequest request)
             return Task.CompletedTask;
         }
 
+        static Task PlatformOpenAsync(OpenFileRequest request, bool openInApp)
```
Comment (dimonovdd) [inline Xamarin.Essentials/Launcher/Launcher.ios.tvos.cs:68]: There is no need to create this method. See @jamesmontemagno  comment.

File: Xamarin.Essentials/Launcher/Launcher.ios.tvos.cs [hunk 4]
```diff
@@ -62,9 +65,60 @@ static Task PlatformOpenAsync(OpenFileRequest request)
...
+        {
+            if (openInApp)
+            {
+                UIApplication.SharedApplication?.KeyWindow?.RootViewController?.
```
Comment (dimonovdd) [inline Xamarin.Essentials/Launcher/Launcher.ios.tvos.cs:72]: use [Platform.GetCurrentUIViewController()](https://github.com/xamarin/Essentials/blob/7218ab88f7fbe00ec3379bd54e6c0ce35ffc0c22/Xamarin.Essentials/Platform/Platform.ios.tvos.watchos.cs#L79)

File: Xamarin.Essentials/Launcher/Launcher.ios.tvos.cs [hunk 5]
```diff
@@ -62,9 +65,60 @@ static Task PlatformOpenAsync(OpenFileRequest request)
...
+
+        public override nint PreviewItemCount(QLPreviewController controller) => 1;
+    }
+
+    public class PreviewItem : QLPreviewItem
+    {
+        public PreviewItem(string filePath, string fileName)
+        {
+            ItemUrl = NSUrl.FromFilename(filePath);
+            ItemTitle = fileName;
```
Comment (dimonovdd) [inline Xamarin.Essentials/Launcher/Launcher.ios.tvos.cs:116]: `fileName` will be used as a window titlel.  Maybe, we should be able to change it.
A file name doesn't always look good

**Review by dimonovdd — COMMENTED** [review]
LGTM.
But I wrote a few comments. Most importantly we need to change the way we get `UIViewController`

**Comment by andrk0p** [comment]
@jamesmontemagno @dimonovdd Thank you for your feedback. I made the changes

**Comment by IngweLand** [comment]
I do not think that QuickLook should be added to the Essentials. It's a dedicated single-platform feature only, while Essentials is supposed to have cross-platform implementations. 
If you still decide to add it, I think you should also add following things:

- Checking if file is supported in QL (https://developer.apple.com/documentation/quicklook/qlpreviewcontroller/1617016-canpreviewitem)
- Disposing of QLPreviewController and its data source

**Comment by jfversluis** [comment]
Hey everyone! Thanks for all the effort and input here! At this point in time we're aiming to not add any new functionality anymore as we're transitioning into .NET MAUI. Also, there seems to be some discussion about if this should be in Essentials because it's very specific to one platform.

If this is still of interest to you, please open a discussion on the .NET MAUI repo and if people seem to be in favor we might be able to port this code there. I'm closing if for here for now. Thank you Andrey for your time here!

**Pull request closed without merging**